### DISCUSSION TITLE
Problem with uploading file

### DISCUSSION THREAD

**Discussion by yanalshoubaki** [livewire/livewire#3385] [1. Help] [answered]
I create a component to create a post, my form has an input file with wire:model="post.post_image" , when I choose an image, I create a function to print the image data but it still gives me null.

my code : 

`
    use WithFileUploads;
    public $post;

    protected $rules = [
        'post.post_image' => 'sometimes|image|max:5000',
        ];

    public function mount ()
    {
        $this->post = new Post;
    }
    public function uploadPostImage(Request $request)
    {
        dd($request->file('post.post_image'));
    }
    public function render ()
    {
        return view('livewire.post.create');
    }

`

html code :
`<div class="sm:col-span-6">
                <input type="file" accept="image/*" wire:model="post.post_image" name="post.post_image" wire:change="uploadPostImage">
            </div>`

errors 👍🏻 

in my console it give me this error

![218857036_2810243302620578_447301210244687888_n](https://user-images.githubusercontent.com/13173362/126388327-8c265d76-933a-442c-97cf-67180b0b5ffe.jpg)

**Comment by ph7jack** [comment]
You don't need call method using **wire:change**. Just use the livewire hooks, like **updatedPostPostImage**

```
public function updatedPostPostImage()
{
    $this->validate();

   //  dd($this->post->post_image));

    $this->post->post_image->store('photos');
}
```

**Reply by yanalshoubaki** [reply]
it's still give me same thing 😥

**Comment by devcircus** [comment, marked as answer]
Don't think you should be using the Request object like that in Livewire. The following is how I handle image uploads. The uploaded file property is a separate entity from the model's image property and should be handled separately.

```php
<?php

namespace App\Http\Livewire\Post;
use Livewire\WithFileUploads;
use Livewire\Component;

class Create extends Component
{
    use WithFileUploads;

    public $post;

    public $postImage;

    protected $rules = [
        'postImage' => 'sometimes|image|max:5000',
    ];

    public function mount ()
    {
        $this->post = new Post;
    }

    public function uploadPostImage()
    {
        dd($this->postImage);
        $url = $this->postImage->store('images');
        $this->post->post_image = $url;
        $this->post->save();
    }

    public function render ()
    {
        return view('livewire.post.create');
    }
```

```html
<div class="sm:col-span-6">
    <input type="file" accept="image/*" wire:model="postImage" name="image">
</div>
````
